test/DynamoCoreTests/PackageDependencyTests.cs

```diff
@@ -0,0 +1,299 @@
+            var deserializedPackageDependencies = JObject["PackageDependencies"];
+            Assert.AreEqual(1, deserializedPackageDependencies.Count());
+            var name = deserializedPackageDependencies.First()["Name"].Value<string>();
+            Assert.AreEqual(package.Name, name);
+            var version = deserializedPackageDependencies.First()["Version"].Value<string>();
+            Assert.AreEqual(package.Version.ToString(), version);
+            var nodes = deserializedPackageDependencies.First()["Nodes"].Values<string>();
+            Assert.AreEqual(package.Nodes.Select(n => n.ToString("N")), nodes);
+        }
+
+        [Test]
+        public void NodeModelPackageDependencyIsCollected()
+        {
+            // Load JSON file graph
+            string path = Path.Combine(TestDirectory, @"core\packageDependencyTests\OneDependentNode_NodeModel.dyn");
+
+            // Assert package dependency is not already serialized to .dyn
+            using (StreamReader file = new StreamReader(path))
+            {
+                var data = file.ReadToEnd();
+                var json = (JObject)JsonConvert.DeserializeObject(data);
+                Assert.IsNull(json["PackageDependencies"]);
+            }
+
+            // Assert package dependency is collected
+            OpenModel(path);
+            var packageDependencies = CurrentDynamoModel.CurrentWorkspace.PackageDependencies;
+            Assert.AreEqual(1, packageDependencies.Count);
+            var package = packageDependencies.First();
+            Assert.AreEqual(new PackageDependencyInfo("Dynamo Samples", new Version("2.0.0")), package);
+            Assert.AreEqual(1, package.Nodes.Count);
+        }
+
+        [Test]
+        public void CustomNodePackageDependencyIsCollected()
+        {
+            // Add "Round Down To Precision" custom node from the "Custom Rounding" package to a new workspace
+            var guid = new Guid("6d1e3caa-780d-40fd-a045-766b3170235d");
+            var customNode = CurrentDynamoModel.CustomNodeManager.CreateCustomNodeInstance(guid);
+            CurrentDynamoModel.AddNodeToCurrentWorkspace(customNode, true);
+
+            // Assert package dependency is collected
+            var packageDependencies = CurrentDynamoModel.CurrentWorkspace.PackageDependencies;
+            Assert.AreEqual(1, packageDependencies.Count);
+            var package = packageDependencies.First();
+            Assert.AreEqual(new PackageDependencyInfo("Custom Rounding", new Version("0.1.4")), package);
+            Assert.AreEqual(1, package.Nodes.Count);
+        }
+
+        [Test]
+        public void PackageDependencyIsCollectedForNewWorkspace()
+        {
+            // Add one node from "Dynamo Samples" package
+            var pi = GetPackageInfo("Dynamo Samples");
+            var node = GetNodeInstance("Examples.NEWBasicExample.Create@double,double,double");
+            CurrentDynamoModel.AddNodeToCurrentWorkspace(node, true);
```

Comment:
@scottmitchell how is this test different from the above test? Did you mean to add a new workspace?